lib/base.js

```diff
@@ -125,6 +125,8 @@ class ConnectionPool extends EventEmitter {
     this.config = config
     this._connected = false
     this._connecting = false
+    // pools are healthy until proven unhealthy
+    this._healthyPool = true
```

Comment:
*personally* I'd only mark the pool as healthy once the initial connection was successful, because a pool that's not connected yet probably shouldn't be healthy

Reply:
Agreed. And `this._isHealthy` would be cleaner, please?

Reply:
So I originally had the pool marked as unhealthy to start, but it complicated the initial connection logic too much (had to basically ignore the unhealthy flag until the first connection was made). It simplified the consuming code a bunch to just make ConnectionPool healthy until proven otherwise.  And thinking about it further, that seemed a more logical choice, since we can't really assume something is unhealthy until we prove it by failing a connect.  Thoughts?

And I'm good with `this._isHealthy`, will update.

Reply:
I would have thought that the initial probe connection would set the pool as healthy - but does that still make the initial set up too complicated?

Reply:
and just to jump onto the bandwagon around variable naming. Our current convention is `_connected`, `_connecting` so `_healthy` actually seems like it would fit best

Reply:
Yeah, I'm not following how marking the pool as healthy only after the initial probe has completed makes it more complex because at the moment we do nothing to act on the healthy flag? it's just there for external consumption.

```js
const pool = new sql.ConnectionPool(config);
pool.healthy == false
await pool.connect()
pool.healthy == true
```

Reply:
ok, let me take a second look at the issue i was running into earlier

Reply:
is there a point to the `connected` flag if it's essentially mirroring the behavior of healthy?  or vice versa?

Reply:
fyi, may be silent on this until next week, some stuff came up

Reply:
I've implemented these thoughts for you and rebased onto the top of the latest #808